server/artifacts/artifact_server.go

```diff
@@ -392,16 +392,20 @@ func (a *ArtifactServer) getArtifactAndDriver(ctx context.Context, nodeId, artif
 	// 2. Defined by Controller configmap
 	// 3. Workflow spec defines artifactRepositoryRef which is a ConfigMap which defines the location
 	// 4. Template defines ArchiveLocation
+	// 5. Inline Template
 
+	var archiveLocation *wfv1.ArtifactLocation
 	templateName := util.GetTemplateFromNode(wf.Status.Nodes[nodeId])
```

Comment:
Was this panicking?

Reply:
Inline template always get empty string from `GetTemplateFromNode`. 
Then it will get `nil` template when `GetTemplateByName`.